mirascope/core/base/call_params.py

```diff
@@ -1,6 +1,78 @@
 """This module contains the type definition for the base call parameters."""
 
-from typing_extensions import TypedDict
+from collections.abc import Callable
+from typing import Any, TypeVar
+
+from typing_extensions import NotRequired, TypedDict
 
 
 class BaseCallParams(TypedDict, total=False): ...  # pragma: no cover
+
+
+_BaseCallParamsT = TypeVar("_BaseCallParamsT", bound=BaseCallParams)
+
+
+class CommonCallParams(TypedDict, total=False):
+    """Common parameters shared across LLM providers.
+
+    Note: Each provider may handle these parameters differently or not support them at all.
+    Please check provider-specific documentation for parameter support and behavior.
+
+    Attributes:
+        temperature: Controls randomness in the output (0.0 to 1.0).
+        max_tokens: Maximum number of tokens to generate.
+        top_p: Nucleus sampling parameter (0.0 to 1.0).
+        frequency_penalty: Penalizes frequent tokens (-2.0 to 2.0).
+        presence_penalty: Penalizes tokens based on presence (-2.0 to 2.0).
+        seed: Random seed for reproducibility.
+        stop: Stop sequence(s) to end generation.
+    """
+
+    temperature: NotRequired[float | None]
+    max_tokens: NotRequired[int | None]
+    top_p: NotRequired[float | None]
+    frequency_penalty: NotRequired[float | None]
+    presence_penalty: NotRequired[float | None]
+    seed: NotRequired[int | None]
+    stop: NotRequired[str | list[str] | None]
+
+
+def convert_params(
```

Comment:
this whole method seems like overkill? it seems like we could just return a dictionary comprehension with a few if/else checks for each provider?

For example, OpenAI would just be this:

```python
mapping = {...}
return {
    mapping[key]: value
    for key, value in common_params.items()
    if key in mapping and mapping[key] is not None
}
```

For gemini we would have this:

```python
mapping = {...}
return {
    mapping[key]: [value] if key == "stop" and isinstance(value, str) else value
    for key, value in common_params.items()
    if key in mapping and mapping[key] is not None
}
```

Seems better also to colocate the provider-specific functionality with that provider's implementation. For example, the `convert_stop_to_list` method is only used by gemini/vertex, but now lives as a base method. Same thing for `wrap_in_config` which is for gemini/vertex. It's not clear why this lives in the `convert_params` function.